src/test/java/org/reaktivity/nukleus/tcp/internal/streams/ServerPartialWriteIT.java

```diff
@@ -117,17 +117,24 @@ public void shouldHandleMultiplePartialWrites() throws Exception
     @BMScript(value="PartialWriteIT.btm")
     public void shouldWriteWhenMoreDataArrivesWhileAwaitingSocketWritable() throws Exception
     {
+        // processData will be called from each of the two data frames. Make the first
+        // do a partial write, then write nothing until handleWrite is called after the
+        // second processData call, when we write everything.
+        AtomicBoolean finishWrite = new AtomicBoolean(false);
         PartialWriteHelper.addWriteResult(5);
-        PartialWriteHelper.addWriteResult(8);
-        PartialWriteHelper.addWriteResult(5);
-        PartialWriteHelper.addWriteResult(5);
+        PartialWriteHelper.setWriteResultProvider((caller) ->
+        {
+            if (caller.equals("processData"))
+            {
+                finishWrite.set(true);
+                return 0;
+            }
+            else
+            {
+                return finishWrite.get() ? null : 0;
+            }
+        });
```

Comment:
Yes it is and that is how the caller parameter is passed in to the preWrite method in PartialWriterHelper. I could have separate setWriteResultProvider methods on PartialWriterHelper to separate the "write called from processData" and "write called from handleWrite" cases, or I could even have separate PartialWriteHelper classes for the two cases. My feeling is that would be make things more complex for the majority of cases which are much simpler.